### PR TITLE
Add editorconfig file and cleanup `devicon.json` bad formatting

### PR DESCRIPTION
**Pull request by BenSouchet**
# Intro
This PR is based on this discussion https://github.com/devicons/devicon/issues/1337 , the idea was, as a first step, to add **EditorConfig** to the project, and I also fix bad formatting on the JSON file.

## What is EditorConfig ?
**EditorConfig** helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. The EditorConfig project consists of a file format for defining coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. EditorConfig files are easily readable and they work nicely with version control systems.

## What is done by EditorConfig
When you work on a file, and for example press the <kbd>tab</kbd> key , you will automatically insert the correct indent type and length, for **Devicon** it's `space of length 4`.
When you save the file, the `charset` defined will be used, whitespace at end of lines (if enable) will be removed. An empty new line will be created at the end of the file if not already present.

## What isn't done
**EditorConfig** is **not** a linter, so the file won't be re-ordered, existing `tab` keys won't be replaced by spaces, or miss-aligned indentation won't be fixed.

**EditorConfig**  is a list of parameters what will be read and used by your current IDE to format the text you are adding, and format the ouput file settings (`charset`, `insert_final_newline`).

# Conclusion
### EditorConfig is one step in the right direction to unified project files formatting
This does not replace a linter, for JSON file [spectral](https://github.com/stoplightio/spectral) could be a good choice. 
Github Hook can be added to lint the JSON file:
https://meta.stoplight.io/docs/spectral/ecaa0fd8a950d-workflows#git-hooks

_Since the "cleaning" of `devicon.json` file can lead to merge conflicts It's maybe a good idea to handle the open pull requests first, then when there is few pull requests still open, this PR can be merge to develop._

### CODE REVIEW

File: devicon.json [hunk 1]
```diff
@@ -3619,7 +3677,7 @@
         ],
         "versions": {
             "svg": [
-		"original",
+        "original",
```
Comment (Snailedlt) [inline devicon.json:3680]: Looks like the indentation is wrong here

File: devicon.json [hunk 2]
```diff
@@ -4768,10 +4832,16 @@
...
+        "tags": [
+            "php",
+            "framework"
+        ],
         "versions": {
-          "svg": ["original", "plain"],
-          "font": ["plain"]
+            "svg": [
+                "original",
+            "plain"
```
Comment (Snailedlt) [inline devicon.json:4842]: Wrong indentation of `"plain"` and `[`

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Nice! This is much needed!
Could only dind three lines that had bad indentation still.

Thanks for making this PR :bow:

File: devicon.json (hunk 2 above)
Comment (BenSouchet) [reply devicon.json]: Fixed 🙂

File: devicon.json (hunk 1 above)
Comment (BenSouchet) [reply devicon.json]: Fixed 🙂

**Review by Snailedlt — APPROVED** [review]
Looks good!
Thanks for your contribution, and a much needed feature :rocket:

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+end_of_line = lf
+insert_final_newline = true
+
+[**.json]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[**.svg]
+trim_trailing_whitespace = false
```
Comment (Panquesito7) [inline .editorconfig:14]: I believe we're missing an endline in here.
```suggestion
trim_trailing_whitespace = false

```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 3 above)
Comment (Snailedlt) [reply .editorconfig:14]: The irony 😅
Did the exact same thing in another peoject, judt with prettier

**Comment by BenSouchet** [comment]
@Snailedlt Hi, due to too much conflicts and issues I decided to create a new branch and so a new PR, check #1364

**Pull request closed without merging**